### PR TITLE
Carlos Created GraphingData class

### PR DESCRIPTION
**Pull request by solarescarlos**
Created a new GraphingData class in graphing_data.py to make our code usable, nice, and pretty, and to pass around the same data object consistently in the backend. 

This class stores a data frame with two columns. Column 1 must be labelled "date" and column 2 must be labelled either "frequency" or "revenue".

Cheers

### CODE REVIEW

**Review by MatthewFrieri — APPROVED** [review]

File: backend/graphing_data.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# This is the GraphingData class.
+# Only a DataFrame with two columns can be store here.
+# Column 1 must be labelled "date" and column 2 must be
+# labelled either "frequency" or "revenue".
+class GraphingData:
+    def __init__(self, data: pd.DataFrame):
+        assert data.shape[1] == 2, "DataFrame must have exactly 2 columns."
+        assert data.columns[1] in ["frequency", "revenue"], "The second column header must be 'frequency' or 'revenue'."
+        self.data = pd.DataFrame
```
Comment (Adventurer-E) [inline backend/graphing_data.py:12]: ?

**Comment by Adventurer-E** [comment]
Do you guys think it's a good idea to make GraphingData a child class of pd.DataFrame? In this way, we would directly uswe the GraphingData variables as DataFrames without having to call getData().

File: backend/graphing_data.py (hunk 1 above)
Comment (solarescarlos) [reply backend/graphing_data.py:12]: Yup, just caught that. Changing now.

File: backend/graphing_data.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import pandas as pd
+
+
+# This is the GraphingData class.
+# Only a DataFrame with two columns can be store here.
+# Column 1 must be labelled "date" and column 2 must be
+# labelled either "frequency" or "revenue".
+class GraphingData:
```
[the remark this answers is not in the document]
Comment (solarescarlos) [reply backend/graphing_data.py:8]: I assume this is for readability. I'll add it in

**Comment by solarescarlos** [comment]
> Do you guys think it's a good idea to make GraphingData a child class of pd.DataFrame? In this way, we would directly uswe the GraphingData variables as DataFrames without having to call getData().

That would be kinda cool for later, I think right now though we should focus on getting this working with the other refactorings.

File: backend/graphing_data.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    data: pd.DataFrame
+
+    def __init__(self, data: pd.DataFrame):
+        assert data.shape[1] == 2, "Data must have exactly 2 columns."
+        assert data.columns[0] == "data", "First column header must be 'data'."
```
Comment (BardyBard) [inline backend/graphing_data.py:13]: make sure that is a datetime.date object, and column name should be "date" not data

**Review by BardyBard — CHANGES_REQUESTED** [review]
Also can you make some simple tests, showing that it works on a properly formatted df and fails on a bad df

**Comment by BardyBard** [comment]
> Do you guys think it's a good idea to make GraphingData a child class of pd.DataFrame? In this way, we would directly uswe the GraphingData variables as DataFrames without having to call getData().

I did not know this was a possibility, that could be a very cool implementation if you can figure out how to do that, I think Paul would be very excited. However, if its complicated for some reason it's not a priority.

**Comment by Adventurer-E** [comment]
> > Do you guys think it's a good idea to make GraphingData a child class of pd.DataFrame? In this way, we would directly uswe the GraphingData variables as DataFrames without having to call getData().
> 
> @Adventurer-E @solarescarlos
> 
> I did not know this was a possibility, that could be a very cool implementation if you can figure out how to do that, I think Paul would be very excited. However, if its complicated for some reason it's not a priority.

Never mind. For now I've implemented ModelInteractor as if GraphingData is not a child of DataFrame.

**Review by Adventurer-E — CHANGES_REQUESTED** [review]
Could you also create a new method GraphingData1.equals(GraphingData2) that checks whether the two contents in GraphingData are equal? It is needed for pytests in other modules. Currently, since GraphingData are objects, a simple == only do address check, which is not what we want.

File: backend/graphing_data.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        assert data.columns[1] in ["frequency", "revenue"], "Second column header must be 'frequency' or 'revenue'."
+        self.data = data
+
+    def getData(self) -> pd.DataFrame:
+        return self.data
```
Comment (Adventurer-E) [inline backend/graphing_data.py:18]: `.copy()`
self.data shouldn't be mutable when it's called.

**Review by Adventurer-E — CHANGES_REQUESTED** [review]

**Comment by Adventurer-E** [comment]
Icannot see flake8 changes under File Changed 🙁

**Comment by solarescarlos** [comment]
> Icannot see flake8 changes under File Changed 🙁

You can't see my style changes?

**Review by avantitandon — COMMENTED** [review]
can't see the flake changes either Carlos

**Comment by solarescarlos** [comment]
> can't see the flake changes either Carlos

You can't see this?
Screenshot (119)

**Review by avantitandon — APPROVED** [review]

**Pull request merged**